wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -92,19 +93,26 @@ class PlatformRepo(val walletApplication: WalletApplication) {
     fun searchUsernames(text: String, userId: String): Resource<List<UsernameSearchResult>> {
         return try {
             // Names.search does support retrieving 100 names at a time if retrieveAll = false
-            var nameDocuments = platform.names.search(text, Names.DEFAULT_PARENT_DOMAIN, true)
+            val nameDocuments = platform.names.search(text, Names.DEFAULT_PARENT_DOMAIN, true)
+
+            val nameDocByUserId = nameDocuments.associateBy({ it.userId }, { it })
+            var userIds = arrayListOf<String>()
+            for (nameDoc in nameDocuments) {
+                userIds.add(nameDoc.userId)
+            }
+            val dashPayDocuments = Profiles(platform).getList(userIds)
```

Comment:
getList will only return the first 100 matches, but there might be a Exception from Platform if `userIds `contains more than 100 items.  We don't have any tests/examples in the DashPay library to know if this is the case.

https://dashplatform.readme.io/docs/reference-query-syntax (see "in")

Reply:
An exception will cause the search method to return an error without any results.  With only about 200 usernames that are registered, we wouldn't encounter this exception.  On Mainnet we would if more than 100 match the search text.